### PR TITLE
Fix test failures in FunctionNode implementation guide

### PR DESCRIPTION
**Pull request by kmaehashi**
Fix `make doctest` error introduced in #4986

### CODE REVIEW

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Doctest still fails.
@kmaehashi Could you fix this?
```
14:51:50 Warning, treated as error:
14:51:50 **********************************************************************
14:51:50 File "guides/functions.rst", line 255, in default
14:51:50 Failed example:
14:51:50     x = Variable(np.random.uniform(-1, 1, (3, 2)).astype(np.float32))
14:51:50     y = Variable(np.random.uniform(-1, 1, (3, 2)).astype(np.float32))
14:51:50     z = Variable(np.random.uniform(-1, 1, (3, 2)).astype(np.float32))
14:51:50     w = MulAdd().apply((x, y, z))
14:51:50     w.grad = np.random.uniform(-1, 1, (3, 2)).astype(np.float32)
14:51:50     w.backward()
14:51:50 Exception raised:
14:51:50     Traceback (most recent call last):
14:51:50       File "/usr/lib/python3.5/doctest.py", line 1321, in __run
14:51:50         compileflags, 1), test.globs)
14:51:50       File "<doctest default[0]>", line 5, in <module>
14:51:50         w.grad = np.random.uniform(-1, 1, (3, 2)).astype(np.float32)
14:51:50     AttributeError: 'tuple' object has no attribute 'grad'
14:51:50 
14:51:51 Makefile:176: recipe for target 'doctest' failed
14:51:51 make: *** [doctest] Error 2
```

**Comment by kmaehashi** [comment]
Jenkins, test this please.

File: docs/source/guides/functions.rst [hunk 1]
```diff
@@ -556,7 +561,7 @@ This is a test example of :func:`functions.relu` function
            x = Variable(np.random.randn(3, 2).astype(np.float32))
            y = F.relu(x)
            y.grad = np.random.randn(3, 2).astype(np.float32)
-           y.backward()
+           y.backward(retain_grad=True)
```
Comment (kmaehashi) [inline docs/source/guides/functions.rst]: Currently `retain_grad=True` is needed due to https://github.com/chainer/chainer/pull/6348

**Comment by toslunar** [comment]
The PR was backported as a part of #6730.  The label `to-be-backported` is removed because the bot does not find it.

**Pull request merged**